src/library/scala/collection/mutable/TreeMap.scala

```diff
@@ -0,0 +1,163 @@
+package scala
+package collection
+package mutable
+
+import scala.collection.generic._
+import scala.collection.mutable.{RedBlackTree => RB}
+
+/**
+ * $factoryInfo
+ *
+ * @define Coll mutable.TreeMap
+ * @define coll mutable tree map
+ */
+object TreeMap extends MutableSortedMapFactory[TreeMap] {
+
+  def empty[A, B](implicit ord: Ordering[A]) = new TreeMap[A, B]()(ord)
+
+  /** $sortedMapCanBuildFromInfo */
+  implicit def canBuildFrom[A, B](implicit ord: Ordering[A]): CanBuildFrom[Coll, (A, B), TreeMap[A, B]] =
+    new SortedMapCanBuildFrom[A, B]
+}
+
+/**
+ * A mutable sorted map implemented using a mutable red-black tree as underlying data structure.
+ *
+ * @param ordering the implicit ordering used to compare objects of type `A`.
+ * @tparam A the type of the keys contained in this tree map.
+ * @tparam B the type of the values associated with the keys.
+ *
+ * @author Rui Gonçalves
+ * @version 2.12
+ * @since 2.12
+ *
+ * @define Coll mutable.TreeMap
+ * @define coll mutable tree map
+ */
+class TreeMap[A, B] private (tree: RB.Tree[A, B])(implicit val ordering: Ordering[A])
```

Comment:
You could mark it `sealed`, rather than `final`. I'll let @Ichoran comment on the specifics of why the immutable version is moving towards finality. I lean towards the "make concrete classes final, or design/document them for extension" school of thought, though admittedly the existing collections classes are more liberal than that.